sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -2527,6 +2527,17 @@ object functions {
     Overlay(src.expr, lit(replaceString).expr, lit(pos).expr, lit(len).expr)
   }
 
+  /**
+   * Overlay the specified portion of `src` with `replaceArray`,
+   *  starting from byte position `pos` of input byte array and proceeding for `len` bytes.
+   *
+   * @group string_funcs
+   * @since 3.0.0
+   */
+  def overlay(src: Column, replaceArray: Array[Byte], pos: Int, len: Int): Column = withExpr {
+    Overlay(src.expr, lit(replaceArray).expr, lit(pos).expr, lit(len).expr)
```

Comment:
let's add it into Python and R side too. This can be done separately

Reply:
Actually, can we use `Column` signature for both `replaceArray` and `replaceString`? We should better stick to `Column`. See the comments above in `functions.scala`.

Reply:
I referenced the function `translate`, `trim`, `regexp_replace` that not use `Column`.

Reply:
Yes, but after that we have landed a change - https://github.com/apache/spark/commit/a133175ffaf8f45d0f2ca731fc0eda8d8f397726. Did you read the comments above in `functions.scala` I mentioned?

Reply:
Yes, I read the comments and use use `Column` signature for both `replaceArray` and `replaceString`.